# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParent.h
# Commit: 43cd1d6eb145
# Full Hash: 43cd1d6eb1459ae16207f4c9396cf5113a7b034a
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-01-11 03:49:34
# Regressor Bug: 1737828
# File Overlap Count: 2
# Description:
#   Bug 1737828 - Avoid the main thread when starting PBackground, r=asuth,ipc-reviewers,necko-reviewers,kershaw,handyman
#   
#   This patch avoids requiring the main thread to create PBackground instances by
#   instead using a background starter TaskQueue, and sending messages from a
#   PBackgroundStarter actor hosted on that thread to the target background thread
# ==============================================================================

diff -r 400d19ddbff7 -r 43cd1d6eb145 ipc/glue/BackgroundParent.h
--- a/ipc/glue/BackgroundParent.h	Mon Jan 10 19:44:23 2022 +0000
+++ b/ipc/glue/BackgroundParent.h	Mon Jan 10 20:09:14 2022 +0000
@@ -37,7 +37,9 @@
 
 namespace ipc {
 
+class BackgroundStarterParent;
 class PBackgroundParent;
+class PBackgroundStarterParent;
 
 template <class PFooSide>
 class Endpoint;
@@ -45,14 +47,15 @@
 // This class is not designed for public consumption beyond the few static
 // member functions.
 class BackgroundParent final {
+  friend class mozilla::ipc::BackgroundStarterParent;
   friend class mozilla::dom::ContentParent;
+  friend class mozilla::net::SocketProcessBridgeParent;
+  friend class mozilla::net::SocketProcessParent;
 
   typedef base::ProcessId ProcessId;
   typedef mozilla::dom::BlobImpl BlobImpl;
   typedef mozilla::dom::ContentParent ContentParent;
   typedef mozilla::ipc::Transport Transport;
-  friend class mozilla::net::SocketProcessBridgeParent;
-  friend class mozilla::net::SocketProcessParent;
 
  public:
   // This function allows the caller to determine if the given parent actor
@@ -81,17 +84,14 @@
 
   static uint64_t GetChildID(PBackgroundParent* aBackgroundActor);
 
-  static bool GetLiveActorArray(PBackgroundParent* aBackgroundActor,
-                                nsTArray<PBackgroundParent*>& aLiveActorArray);
-
  private:
   // Only called by ContentParent for cross-process actors.
-  static bool Alloc(ContentParent* aContent,
-                    Endpoint<PBackgroundParent>&& aEndpoint);
+  static bool AllocStarter(ContentParent* aContent,
+                           Endpoint<PBackgroundStarterParent>&& aEndpoint);
 
   // Called by SocketProcessBridgeParent and SocketProcessParent for
   // cross-process actors.
-  static bool Alloc(Endpoint<PBackgroundParent>&& aEndpoint);
+  static bool AllocStarter(Endpoint<PBackgroundStarterParent>&& aEndpoint);
 };
 
 // Implemented in BackgroundImpl.cpp.